core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -95,6 +95,19 @@ sealed trait ConfigCursor {
   def asMap: Either[ConfigReaderFailures, Map[String, ConfigCursor]] =
     asObjectCursor.right.map(_.map)
 
+  final def at(pathSegment: PathSegment): Either[ConfigReaderFailures, ConfigCursor] = {
+    pathSegment match {
+      case PathSegment.Key(k) => this.asObjectCursor.right.flatMap(_.atKey(k))
+      case PathSegment.Index(i) => this.asListCursor.right.flatMap(_.atIndex(i))
+    }
+  }
+
+  final def at(pathSegments: PathSegment*): Either[ConfigReaderFailures, ConfigCursor] = {
```

Comment:
Can you please also add Scaladoc for these methods?

Reply:
Sure thing.